### PR TITLE
elixir-ls: compile using elixir 1.14.2(latest)

### PR DESCRIPTION
**Pull request by robsonpeixoto**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/elixir-ls.rb [hunk 1]
```diff
@@ -41,5 +41,6 @@ def install
 
     output = pipe_output("#{bin}/elixir-ls", input, 0)
     assert_match "Content-Length", output
+    assert_match "ElixirLS built with elixir \\\"1.14.2\\\"", output
```
[the remark this answers is not in the document]
Comment (robsonpeixoto) [reply Formula/elixir-ls.rb:44]: I just follow the same idea of this PR https://github.com/Homebrew/homebrew-core/pull/108141

[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/elixir-ls.rb:44]: ok, commenting there as well.

**Comment by chenrui333** [comment]
This should be covered by https://github.com/Homebrew/homebrew-core/actions/runs/3566211042

File: Formula/elixir-ls.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/elixir-ls.rb:44]: Do you have any issue report about why there is strong need to recompile the formula?

**Comment by chenrui333** [comment]
The action build failures are related to github outage.

**Comment by robsonpeixoto** [comment]
Sorry, I'm busy now to check if exists erros with the current elixir-ls.

But in my experience the elixir-ls works better when is compiled with the elixir and erlang version that are being used.

Ate there a way to flag a package to rebuild when a build_dependency updates?

**Pull request closed without merging**